Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp

```diff
@@ -2530,6 +2533,22 @@ JS::ThrowCompletionOr<JS::NonnullGCPtr<JS::Object>> @constructor_class@::constru
         }
 
         constructor_generator.append(R"~~~(
+    if (prototype.is_object()) {
+        // 3.3. If Type(prototype) is not Object, then:
+        if (!prototype.is_object()) {
```

Comment:
It looks like these if statements need to be reordered. Currently this statement will never be executed given the if statement above.